autoload/elixir/indent.vim

```diff
@@ -62,6 +62,14 @@ function! elixir#indent#deindent_ending_symbols(ind, line)
 endfunction
 
 function! elixir#indent#deindent_keywords(ind, line)
+  echom "> pair start ". s:PAIR_START
+  echom "> pair middle ". s:PAIR_MIDDLE
+  echom "> pair end ". s:PAIR_END
+  echom "> pair skip ". s:PAIR_SKIP
+
+  " NOTE: cursor is not being properly set here which causes this to
+  " incorrectly handle nesting
+  echom "> cursor => " . line(".") ", " . col(".")
   if a:line.current.text =~ s:DEINDENT_KEYWORDS
     let bslnum = searchpair(
```

Comment:
@kassio @gasparch not sure if you have any ideas around this: 340b281ad6b38513337465337677851c407d608d fixes this issue however the `searchpair` call is not always correct.

It looks like the trailing `\zs` in `PAIR_END` causes the following tests to still fail. Any thoughts why this is the case? Removing it causes several other tests to fail

```
Failures:

  1) Indenting anonymous functions fn with comment with "end"
     Failure/Error:
       expect(<<~EOF).to be_elixir_indentation
       fun3 = fn
         (:foo) ->
           :foo
           :asd # end
       end
       EOF

       @@ -2,5 +2,5 @@
          (:foo) ->
            :foo
            :asd # end
       -end
       +  end
     # ./spec/indent/anonymous_functions_spec.rb:111:in `block (2 levels) in <top (required)>'

  2) Indenting anonymous functions fn with string with "end"
     Failure/Error:
       expect(<<~EOF).to be_elixir_indentation
       fun2 = fn
         (:foo) ->
           :foo
           "string with end"
       end
       EOF

       @@ -2,5 +2,5 @@
          (:foo) ->
            :foo
            "string with end"
       -end
       +  end
     # ./spec/indent/anonymous_functions_spec.rb:101:in `block (2 levels) in <top (required)>'

  3) Indenting documentation with end keyword
     Failure/Error:
       expect(<<~EOF).to be_elixir_indentation
       defmodule Test do
         @doc """
         end
         """
       end
       EOF

       @@ -2,5 +2,5 @@
          @doc """
          end
          """
       -end
       +  end
     # ./spec/indent/documentation_spec.rb:7:in `block (2 levels) in <top (required)>'

Finished in 39.72 seconds (files took 0.12091 seconds to load)
200 examples, 3 failures, 5 pending

Failed examples:

rspec ./spec/indent/anonymous_functions_spec.rb:110 # Indenting anonymous functions fn with comment with "end"
rspec ./spec/indent/anonymous_functions_spec.rb:100 # Indenting anonymous functions fn with string with "end"
rspec ./spec/indent/documentation_spec.rb:6 # Indenting documentation with end keyword
```